### PR TITLE
Split common into common and types

### PR DESCRIPTION
**Pull request by skejserjensen**
This PR splits `modelardb_common` into `modelardb_common` and `modelardb_types` so `ModelarDbError` can be used without a lot of dependencies in preparation for working on #246.

### CODE REVIEW

File: crates/modelardb_types/Cargo.toml [hunk 1]
```diff
@@ -0,0 +1,27 @@
+# Copyright 2023 The ModelarDB Contributors
```
Comment (chrthomsen) [inline crates/modelardb_types/Cargo.toml:1]: 2024 ?

File: crates/modelardb_types/src/lib.rs [hunk 2]
```diff
@@ -0,0 +1,22 @@
+/* Copyright 2023 The ModelarDB Contributors
```
Comment (chrthomsen) [inline crates/modelardb_types/src/lib.rs:1]: 2024 ?

**Review by chrthomsen — APPROVED** [review]

File: crates/modelardb_types/Cargo.toml (hunk 1 above)
Comment (skejserjensen) [reply crates/modelardb_types/Cargo.toml:1]: Fixed.

File: crates/modelardb_types/src/lib.rs (hunk 2 above)
Comment (skejserjensen) [reply crates/modelardb_types/src/lib.rs]: Fixed.

**Review by CGodiksen — APPROVED** [review]

**Pull request merged**